controllers/toolchaincluster/toolchaincluster_controller_test.go

```diff
@@ -209,3 +300,27 @@ func prepareReconcile(toolchainCluster *toolchainv1alpha1.ToolchainCluster, cl *
 	}
 	return controller, req
 }
+
+func withStatus(conditions ...toolchainv1alpha1.Condition) toolchainv1alpha1.ToolchainClusterStatus {
+	return toolchainv1alpha1.ToolchainClusterStatus{
+		Conditions: conditions,
+	}
+}
+func assertClusterStatus(t *testing.T, cl client.Client, clusterName string, clusterConds ...toolchainv1alpha1.Condition) {
+	tc := &toolchainv1alpha1.ToolchainCluster{}
+	err := cl.Get(context.TODO(), test.NamespacedName("test-namespace", clusterName), tc)
+	require.NoError(t, err)
+	assert.Len(t, tc.Status.Conditions, len(clusterConds))
+ExpConditions:
+	for _, expCond := range clusterConds {
+		for _, cond := range tc.Status.Conditions {
+			if expCond.Type == cond.Type {
+				assert.Equal(t, expCond.Status, cond.Status)
+				assert.Equal(t, expCond.Reason, cond.Reason)
+				assert.Equal(t, expCond.Message, cond.Message)
+				continue ExpConditions
+			}
+		}
+		assert.Failf(t, "condition not found", "the list of conditions %v doesn't contain the expected condition %v", tc.Status.Conditions, expCond)
+	}
```

Comment:
I believe that you can simplify this part of the code since we use only one condition type